extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/GeneratedStaticResourcesProcessor.java

```diff
@@ -0,0 +1,90 @@
+package io.quarkus.vertx.http.deployment;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.GeneratedResourceBuildItem;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
+import io.quarkus.vertx.http.deployment.spi.AdditionalStaticResourceBuildItem;
+import io.quarkus.vertx.http.deployment.spi.GeneratedStaticResourceBuildItem;
+import io.quarkus.vertx.http.runtime.GeneratedStaticResourcesRecorder;
+import io.quarkus.vertx.http.runtime.RouteConstants;
+import io.quarkus.vertx.http.runtime.handlers.DevClasspathStaticHandler;
+
+/**
+ * {@link GeneratedStaticResourcesProcessor} is responsible for dealing {@link GeneratedStaticResourceBuildItem}
+ * creating a {@link DevClasspathStaticHandler} to handle all static resources
+ * generated from extensions through {@link GeneratedStaticResourceBuildItem} build item.
+ */
+public class GeneratedStaticResourcesProcessor {
+
+    private static final int ROUTE_ORDER = RouteConstants.ROUTE_ORDER_BEFORE_DEFAULT + 60;
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
```

Comment:
I think you should separate this in two buildsteps, one for creating the different buildItems and another for the runtime init that uses the recorder. You can choose to create the Map in a new internal BuildItem or create it in runtime init